shell/platform/glfw/flutter_glfw.cc

```diff
@@ -149,10 +149,54 @@ struct FlutterDesktopPluginRegistrar {
 
 // State associated with the messenger used to communicate with the engine.
 struct FlutterDesktopMessenger {
+  void AddRef() { ref_count_.fetch_add(1); }
+
+  void Release() {
+    int32_t old_count = ref_count_.fetch_sub(1);
+    if (old_count <= 1) {
+      delete this;
+    }
+  }
+
+  FlutterDesktopEngineState* GetEngine() const { return engine_; }
+  void SetEngine(FlutterDesktopEngineState* engine) {
+    std::scoped_lock lock(mutex_);
+    engine_ = engine;
+  }
+
+  std::mutex& GetMutex() { return mutex_; }
+
+ private:
```

Comment:
Should this also have a `FML_DISALLOW_COPY_AND_ASSIGN(FlutterDesktopMessenger);`?

Reply:
Yep, thanks, done.